qt/applications/workbench/workbench/plotting/figuremanager.py

```diff
@@ -312,6 +316,11 @@ def show(self):
                 not self.canvas.figure.get_axes()[0].is_waterfall():
```

Comment:
This module is getting pretty heavy - as this and `set_up_color_selector_toolbar_button` is toolbar related perhaps we could move this to `toolbar.py`? and maybe add a `update(fig)` method to keep this all together?